scipy/stats/_continuous_distns.py

```diff
@@ -782,12 +788,41 @@ def _pdf(self, x, c, d):
         # burr.pdf(x, c, d) = c * d * x**(-c-1) * (1+x**(-c))**(-d-1)
         return c * d * (x**(-c - 1.0)) * ((1 + x**(-c))**(-d - 1.0))
 
+    def _logpdf(self, x, c, d):
+        return (np.log(c) + np.log(d) + (-c - 1.0) * np.log(x) +
+                 (-d - 1.0) * np.log1p(x**(-c)))
+
     def _cdf(self, x, c, d):
         return (1 + x**(-c))**(-d)
 
+    def _logcdf(self, x, c, d):
+        return -d * np.log1p(x**(-c))
+
+    def _sf(self, x, c, d):
+        return 1 - (1 + x**(-c))**(-d)
+
     def _ppf(self, q, c, d):
         return (q**(-1.0/d) - 1)**(-1.0/c)
 
+    def _stats(self, c, d):
+        e1 = self._munp(1, c, d)
+        e2 = self._munp(2, c, d)
+        e3 = self._munp(3, c, d)
+        e4 = self._munp(4, c, d)
+        mu = np.where(c > 1.0, e1, np.nan)
+        mu2_if_c = e2 - mu**2
+        mu2 = np.where(c > 2.0, mu2_if_c, np.nan)
+        g1 = _lazywhere(c > 3.0, (c, e1, e2, e3, mu2_if_c),
+             lambda c, e1, e2, e3, mu2_if_c:
+             (e3 - 3 * e2 * e1 + 2 * e1**3) / np.sqrt((mu2_if_c)**3),
```

Comment:
Nit: I feel grouping operators by priority a bit more readable. Like so, `e3 - 3*e2*e1 + ...`. No need to change it here though.

Reply:
These are the raw moments, e.g. expected value of `x`, `E(X) = e1` second raw moment of `x`, `E(X^2) = e2` etc.
Do I understand correctly that your recommendation is to do the calculations directly on the  mean, variance, skewness, and kurtosis? I think that would be more code but I could do it if needed, would a comment here defining `e1`, `e2` etc help readability? Something like
```
#ek is the kth raw moment, e.g. e1 is the mean e2-e1**2 is variance etc.  
```

Reply:
Nope, my comment is way more trivial, and is only about whitespace.